src/net/tls.cc

```diff
@@ -1381,23 +1381,26 @@ class session : public enable_lw_shared_from_this<session> {
             gnutls_transport_set_errno(*this, EAGAIN);
             return -1;
         }
-        if (_output_pending.failed()) {
-            // copy exception for on_internal_error
-            auto ex = _output_pending.get_exception();
-            _output_pending = make_exception_future<>(ex);
-            on_internal_error(seastar_logger, format("tls: session: vec_push: unhandled pending output error: {}", ex));
-        }
         try {
+            ssize_t n;
```

Comment:
what else case?
If pending_error is false, we set n, otherwise we rethrow the exception and never get to return n.
I can set to it 0, but I actually want the compiler to warn us if n is returned by some path where it's uninitialized, since I don't want the function to return 0 inadvertently either.

Reply:
Maybe I missed something... But if there are really two different paths here, one which sets and returns n, and one which doesn't, can't it be written like the old code where n is declared only in the first path?

Reply:
the problem is that the good path is broken into two blocks hence n was moved out a scope.